### PR TITLE
Add ostree update scripts.

### PR DESCRIPTION
**Pull request by petedyerarm**
Added scripts for the generation of ostree static delta images.

ostree-delta.py does all of the heavy lifting.

If 2 repos are provided it pulls one into the other and creates the
static delta between them.
If a single repo is provided then it produces the static delta between
the last 2 commits.
Optionally specific shas can be provided to create the diffs between
specific commits.

The createOSTreeUpgrade.sh script mounts the ostree repos from the old
and new wic files and then calls upon ostree-delta.py to produce the
static delta between these two repos.

### CODE REVIEW

**Comment by JanneKiiskila** [comment]
@ygoyal18  @x85446  - please review.

File: ostree/Docker/config-funcs.sh [hunk 1]
```diff
@@ -0,0 +1,350 @@
+#!/bin/bash
+
+# Copyright (c) 2019, Arm Limited and Contributors. All rights reserved.
```
Comment (JanneKiiskila) [inline ostree/Docker/config-funcs.sh:3]: Copyright (c) 2021.

File: ostree/Docker/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Copyright (c) 2019, Arm Limited and Contributors. All rights reserved.
```
Comment (JanneKiiskila) [inline ostree/Docker/entrypoint.sh:3]: Same here, copyright 2021 Pelion.

File: ostree/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# OSTree-update-scripts
+Build scripts and tools for the OSTree firmware images.  
+To create the images, ostree must either be installed on the system, or within a Docker container. 
+There are two ways of creating an update image:   
+   - between two wic images.
+   - between ostree repositories.
```
Comment (JanneKiiskila) [inline ostree/README.md:6]: OSTree

File: ostree/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# OSTree-update-scripts
+Build scripts and tools for the OSTree firmware images.  
+To create the images, ostree must either be installed on the system, or within a Docker container. 
+There are two ways of creating an update image:   
+   - between two wic images.
+   - between ostree repositories.   
+   
+These artifacts are produced during the yocto build.
```
Comment (JanneKiiskila) [inline ostree/README.md:8]: Yocto

File: ostree/createOSTreeUpgrade.sh [hunk 5]
```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+
+# Copyright (c) 2020, Arm Limited and affiliates.
```
Comment (JanneKiiskila) [inline ostree/createOSTreeUpgrade.sh:3]: 2021 Pelion

File: ostree/createOSTreeUpgrade.sh [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+# SPDX-License-Identifier: Apache-2.0
```
Comment (JanneKiiskila) [inline ostree/createOSTreeUpgrade.sh:4]: Others were BSD-3, this is now Apache 2.0? Should all of these be Apache 2.0?

File: ostree/ostree-delta-docker.sh [hunk 7]
```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+
+# Copyright (c) 2018-2019, Arm Limited and Contributors. All rights reserved.
```
Comment (JanneKiiskila) [inline ostree/ostree-delta-docker.sh:3]: 2018-2021

File: ostree/ostree-delta.py [hunk 8]
```diff
@@ -0,0 +1,446 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019, Arm Limited and Contributors. All rights reserved.
```
Comment (JanneKiiskila) [inline ostree/ostree-delta.py:3]: 2021, Pelion

File: ostree/Docker/entrypoint.sh (hunk 2 above)
Comment (JanneKiiskila) [reply ostree/Docker/entrypoint.sh:3]: Should be also Apache 2.0 (generically, all of these) - top level header file on this repo is Apache 2.0 license. No BSD-3 ways @petedyerarm.

**Comment by cmonr** [comment]
Added reviewers to make sure this didn't get lost.

I do remember @petedyerarm you saying that you didn't have access to add them.

**Review by x85446 — APPROVED** [review]

**Pull request merged**